### PR TITLE
Native AudioDecoder and VideoDecoder

### PR DESCRIPTION
**Pull request by tomspilman**
This is a follow-up to #8309 and part of the work on #8194.

This implements two new native APIs `AudioDecoder` and `VideoDecoder`.

These native APIs live under the `Video` and `Song` classes.

The design aims to let the native code handle the work of detecting the media format, selecting the correct decoder, process the media data, and return frames to C#.

With `AudioDecoder` this means C# keeps a thread and pumps signed 16bit PCM data to a native `Voice`.

With `VideoDecoder` a C# thread gathers `Texture`s and calls an `AudioDecoder` to process audio.

There is no public implementation of audio or video decoders yet.   The only implementation currently is for a native audio format of Nintendo Switch.  The video decoder is totally experimental and untested, but i think it will generally work.

I expect it be a good project for someone or should be made into bounties for different audio/video codecs.

As always feedback is welcome!

### CODE REVIEW

**Review by dellis1972 — APPROVED** [review]

File: MonoGame.Framework/Platform/Media/Song.iOS.cs [hunk 1]
```diff
@@ -193,15 +192,6 @@ private Genre PlatformGetGenre()
             return this.genre;
         }
 
-        private TimeSpan PlatformGetDuration()
```
Comment (tomspilman) [inline MonoGame.Framework/Platform/Media/Song.iOS.cs:196]: @mrhelmut i fixed these cases so it always assigns to the common _duration type.

File: MonoGame.Framework/Platform/Media/Song.Android.cs [hunk 2]
```diff
@@ -183,11 +178,6 @@ private bool PlatformIsRated()
             return false;
         }
 
-        private string PlatformGetName()
```
Comment (tomspilman) [inline MonoGame.Framework/Platform/Media/Song.Android.cs:186]: Removed these as well.

For all platforms it defaults to the file name unless the platform code changes it during Song initialization.

**Comment by espinosaluis** [comment]
After 5e918e9 commit, music no longer plays on Android devices.  This was released on 3.8.2.1520-develop
Music plays when reverting back to 3.8.2.1362-develop
Error:
`Java.IO.FileNotFoundException: menu
   at Java.Interop.JniEnvironment.InstanceMethods.CallObjectMethod(JniObjectReference instance, JniMethodInfo method, JniArgumentValue* args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/net7.0/JniEnvironment.g.cs:line 19992
   at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeAbstractObjectMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:line 848
   at Android.Content.Res.AssetManager.OpenFd(String fileName) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.Content.Res.AssetManager.cs:line 308
   at Microsoft.Xna.Framework.Media.Song.Play(Nullable1 startPosition)
   at Microsoft.Xna.Framework.Media.MediaPlayer.PlatformPlaySong(Song song, Nullable1 startPosition)
   at Microsoft.Xna.Framework.Media.MediaPlayer.PlaySong(Song song, Nullable1 startPosition)
   at Microsoft.Xna.Framework.Media.MediaPlayer.Play(Song song, Nullable1 startPosition)
   at Microsoft.Xna.Framework.Media.MediaPlayer.Play(Song song)
   at OldWays.Music.Play() in C:\Programming\Visual C#\Work\OldWays\Music.cs:line 182
  --- End of managed Java.IO.FileNotFoundException stack trace ---
java.io.FileNotFoundException: menu
	at android.content.res.AssetManager.nativeOpenAssetFd(Native Method)
	at android.content.res.AssetManager.openFd(AssetManager.java:934)

  --- End of managed Java.IO.FileNotFoundException stack trace ---
java.io.FileNotFoundException: menu
	at android.content.res.AssetManager.nativeOpenAssetFd(Native Method)
	at android.content.res.AssetManager.openFd(AssetManager.java:934)`

**Comment by espinosaluis** [comment]
@tomspilman Let me know if you need some additional information.

**Comment by dellis1972** [comment]
@espinosaluis can you open a new issue https://github.com/MonoGame/MonoGame/issues/new/choose and reference this commit in it please. That way we can track when the fix lands.

**Comment by espinosaluis** [comment]
> @espinosaluis can you open a new issue https://github.com/MonoGame/MonoGame/issues/new/choose and reference this commit in it please. That way we can track when the fix lands.

Thanks, Dean. I just did it.
https://github.com/MonoGame/MonoGame/issues/8559

**Pull request merged**